package com.liceo.mysocial.ui

import androidx.lifecycle.ViewModel
import androidx.lifecycle.viewModelScope
import com.liceo.mysocial.data.ThemeRepository
import kotlinx.coroutines.flow.*
import kotlinx.coroutines.launch

class ThemeViewModel(private val repo: ThemeRepository) : ViewModel() {

    val isDarkTheme: StateFlow<Boolean> = repo.isDarkTheme
        .stateIn(viewModelScope, SharingStarted.WhileSubscribed(5_000), false)

    fun onThemeChanged(dark: Boolean) {
        viewModelScope.launch {
            repo.setDarkTheme(dark)
        }
    }
}
